### ISSUE TITLE
Regarding the BG Database Connector Issue

### ISSUE THREAD

**Issue by PRISHIta123** [carboncopies/BrainGenix#14]
As the MySQL database interface code (main.py) is returning the following error: 

`mysql.connector.errors.NotSupportedError: Authentication plugin 'caching_sha2_password' is not supported`

a possible solution maybe to try to install the mysql-connector-python package instead of importing mysql.connector in ./System/Database/DatabaseInterface.py" as suggested in the second answer here: https://stackoverflow.com/questions/50557234/authentication-plugin-caching-sha2-password-is-not-supported

This package could be installed by running the install.sh file directly, without having to separately do this later.

**Comment by datacrystals** [comment]
@bkl4114 

I'm still having issues with sha2 password caching. Can you take a look at this?

```
Connecting to MySQL database...
Authentication plugin 'caching_sha2_password' is not supported
Authentication plugin 'caching_sha2_password' is not supported
Traceback (most recent call last):
  File "DbConnectTest.py", line 32, in <module>
    DbQuery.UserLogin('bleu', '123456')
  File "/home/BrainGenix/System/DbQuery.py", line 57, in UserLogin
    cursor.close()
UnboundLocalError: local variable 'cursor' referenced before assignment
```

**Comment by bkl4114** [comment]
I'm using Pycharm to debug, so it probably auto installs plug-ins for me. I'll take a look this weekend.

**Comment by bkl4114** [comment]
Good news and bad news, when I changed to Python 3.8 I was able to recreate the problem on my local, but no matter what I do I can't get around the error.

**Comment by bkl4114** [comment]
@datacrystals , I have a branch called Issue-14-fix-db. When I wiped the slate clean, then upgraded to Python 3.8 and reinstalled the plugin for mysql-connector-python it seems to fix the issue, but it's about 3800 new files. Would you please try my new branch on the server and see if that fixes the sha2 error? If it does, I'll put in a pull request to merge with master.

**Comment by datacrystals** [comment]
@bkl4114 I've sadly still got the same issue...

```
tliao@bg-turing-0:/home/BrainGenix/System$ python3 DbConnectTest.py 
Connecting to MySQL database...
Authentication plugin 'caching_sha2_password' is not supported
Authentication plugin 'caching_sha2_password' is not supported
Traceback (most recent call last):
  File "DbConnectTest.py", line 32, in <module>
    DbQuery.UserLogin('bleu', '123456')
  File "/home/BrainGenix/System/DbQuery.py", line 57, in UserLogin
    cursor.close()
UnboundLocalError: local variable 'cursor' referenced before assignment
```
You should be able to login to the server as well and test it yourself if you need to.
Is merging all of the packages in the venv folder a good idea? Seems like we shouldn't have to do that...

**Comment by bkl4114** [comment]
That's all stuff that Pycharm did when I upgraded to 3.8 and it made things work on my local, but if it doesn't work on the server I won't do a pull request.

**Comment by datacrystals** [comment]
Hmm, alright I did try something though. If i use a different MySQL module, I was able to get it to work (Yay!) I'm going to switch the dependencies to pymysql rather than mysql connector. Additionally, i just realized that the database bgdb didn't exist on the server, so that may be a reason why it didn't work. Luckily, it seems that your generateBrianDb.sql file worked for me when i ran "SOURCE ../../Database/generateBrianDb.sql"

For now, though, I'll go and resolve these changes and then close this issue.
